Src/Base/AMReX_FabArrayBase.cpp

```diff
@@ -1244,7 +1244,10 @@ FabArrayBase::getTileArray (const IntVect& tilesize) const
 #pragma omp critical(gettilearray)
 #endif
     {
-	BL_ASSERT(getBDKey() == m_bdkey);
+      if(getBDKey() != m_bdkey)
```

Comment:
Could we put BL_ASSERT back?  If one compile with `USE_ASSERTION=TRUE`, assertion will be turned on without having to run in debug mode.